### DISCUSSION TITLE
Choosing the respondent in chats mode with multiple opponnents

### DISCUSSION THREAD

**Discussion by OcelotTheOcelot** [LostRuins/koboldcpp#433] [Ideas]
In (Auto) mode, where the AI Name box is clear, the order of respondents stays the same. In "Op1||$||Op2" case, the opponent seems to be randomly chosen every time, but there are cases when the user would like a specific one to answer (e.g. when you ask one character and the other starts responding for them instead).
I think it would be a neat feature if the user could end their line with "This is my question \n Op1:" and the UI would recognise it without appending a random opponent's name to the prompt like it seems to be doing right now.

**Comment by LostRuins** [comment]
Yeah opponents are randomly selected currently. this idea was mentioned before - but a reliable way to do this is harder than it looks, since mentioning a character is kind of contextual.

**Reply by OcelotTheOcelot** [reply]
It seems like the UI in Auto mode already analyzes the text for opponents, inferring them by the "Name:" construction. Why not just check the prompt for having "Some opponent's name:" at the end (since the list of opponents is already inferred as I understand it) so that the UI doesn't append "\nRandom opponent's name:" at the end? May be even prepend it with "\n" for a single-line prompt box in the "Aesthetic" mode.

**Reply by LostRuins** [reply]
Not a bad idea, will do that (or something similar)

**Reply by LostRuins** [reply]
This will be added in the next release. It is already live on lite.koboldai.net
